lib/internal/streams/readable.js

```diff
@@ -196,7 +196,8 @@ function Readable(options) {
   Stream.call(this, options);
 
   destroyImpl.construct(this, () => {
-    maybeReadMore(this, this._readableState);
+    if (!options || !options.manualStart)
+      maybeReadMore(this, this._readableState);
```

Comment:
This needs some tests that are specific to streams. How is a user of Stream be able to use this?

Reply:
@mcollina @joyeecheung @ronag @mscdex 

Attaching a `data` handler will start the stream.
I simply recreated the semantics of the non-documented option in `net.Socket` - which is what the thread bootstrap code uses.
Its only user at the moment is this thread bootstrap code for the `stdin` stream. The `net.Socket` option has a second use case in `TLSWrap` - but I am not sure if it actually works - because this wouldn't happen : <https://github.com/nodejs/node/issues/35475>

The reason I did in `Readable` is simply that the reading is started from here for all `Readable` except `net.Socket` which has its own startup code

To move this to `ReadStream` I would need to reimplement part of this code and it will be a much more invasive change

So two options:
* Keep both `manualStart` options as hidden non-documented options, eventually replacing them with `Symbol` names
* Document them, unit test them and make them official

It is also worth noting that there are two other `net.Socket` specific options, both non-documented, `pauseOnStart` and  `pauseOnConnect` - which have different semantics - the bootstrap code being made for `manualStart`